src/java.desktop/macosx/classes/sun/lwawt/LWWindowPeer.java

```diff
@@ -289,6 +289,15 @@ protected void setVisibleImpl(final boolean visible) {
         updateFocusableWindowState();
         super.setVisibleImpl(visible);
         // TODO: update graphicsConfig, see 4868278
+        if (visible) {
+            // Set correct background for a window before making it visible.
+            // It is necessary to eliminate a flashing effect
+            // (when initial background color replaced with the content layer's color)
+            Color color = getBackground();
+            if (color != null) {
+                platformWindow.setBackground(color);
```

Comment:
If it skips the texture property I guess it will break it?
I think the easiest solution for this issue is to add a new method to the LWWindowPeer:
 ```
   public void setBackground(final Color c) {
        Color oldBg = getBackground();
        if (oldBg == c || (oldBg != null && oldBg.equals(c))) {
            return;
        }
        super.setBackground(c);
        updateOpaque();
    }
}
```

Reply:
> If it skips the texture property I guess it will break it?
I've not noticed it in my testing. Do you mean particular tests? But anyway your suggestion looks more elegant and it works in my scenarios, also I've run test-jdk_desktop_part1 with it. No related problem was found.

Reply:
The CPlatformWindow.setOpaque() changes the background of the native window only if the texture property is not set. So if the code will change the color directly somewhere then the texture property will be broken. Not sure that we have a test for that.
Please update the copyright date.

Reply:
@mrserb updated.